src/pytorch_lightning/callbacks/progress/tqdm_progress.py

```diff
@@ -375,9 +379,7 @@ def convert_inf(x: Optional[Union[int, float]]) -> Optional[Union[int, float]]:
     return x
 
 
-def _update_n(bar: _tqdm, current: int, refresh_rate: int) -> None:
+def _update_n(bar: _tqdm, value: int) -> None:
     if not bar.disable:
-        total = bar.total
-        leftover = current % refresh_rate
-        advance = leftover if (current == total and leftover != 0) else refresh_rate
-        bar.update(advance)
+        bar.n = value
```

Comment:
and keep this as it is? to make sure `miniters` is considered if set by user using tqdm options.
I don't know a lot about tqdm internals but manually setting just the value of `n` skips the update for `last_print_n` and `last_print_t`.

Reply:
The previous way with update() would increase the bar by increments, but this is not correct when resuming with our counters. We can look for a better solution but this is the only one I found that worked.

Reply:
just want to make sure the old reported issues don't come up again since there's no way to test this in the CI to avoid regression. Will explore more on this.. let's merge this then for now.

Reply:
Yes I saw the issues that the previous PR fixed. I checked [this](https://github.com/Lightning-AI/lightning/issues/12875) yesterday and it was also caused by not resetting tqdm.initial to 0.